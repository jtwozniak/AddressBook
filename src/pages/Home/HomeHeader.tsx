import React from "react"
import styled from "styled-components"
import { Search } from "./Search"

const Div = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
`
const MainDiv = styled.div`
  display: flex;
  flex-direction: row;
  width: 1000px;
  justify-content: space-between;
  padding-bottom: 2em;

  align-self: center;

  h3 {
    display: inline-block;
    margin: 0 1em;
  }
`

type Props = {
  page: number
  usersCount: number
  filteredUsersCount: number
  onSearchChange: (v: string) => void
}

// Search bar + some extra stats to see what is the state fetched data
export const HomeHeader = ({
  page,
  usersCount,
  onSearchChange,
  filteredUsersCount,
}: Props) => {
  return (
    <MainDiv>
      <Search onChange={onSearchChange} />

      <Div>
        <h3>Filtered users: {filteredUsersCount}</h3>
        <h3>Fetched users: {usersCount}</h3>
        <h3>Pages: {page}</h3>
      </Div>
    </MainDiv>
  )
}
