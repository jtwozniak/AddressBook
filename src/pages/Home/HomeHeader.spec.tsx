import React from "react"
import { render, fireEvent } from "@testing-library/react"
import { HomeHeader } from "./HomeHeader"

describe("Homeheader", () => {
  const props = {
    page: 1,
    usersCount: 10,
    filteredUsersCount: 5,
    onSearchChange: jest.fn(),
  }

  it("snapshot", () => {
    const { container } = render(<HomeHeader {...props} />)
    expect(container).toMatchSnapshot()
  })

  it("on search change", () => {
    let onSearch = ""
    const onSearchChange = jest.fn((s: string) => {
      onSearch = s
    })

    const rest = { ...props, onSearchChange }
    const { getByLabelText } = render(<HomeHeader {...rest} />)

    const InputText = "TestSearch"
    const input = getByLabelText("search-input")
    fireEvent.change(input, {
      target: { value: InputText },
    })

    expect(onSearchChange).toBeCalledTimes(1)
    expect(onSearch).toBe(InputText)
  })
})
