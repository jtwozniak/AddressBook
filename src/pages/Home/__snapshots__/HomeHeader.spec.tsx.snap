// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homeheader snapshot 1`] = `
.c1 {
  margin: 0 1em;
  width: 250px;
  background-color: #042653;
  color: white;
  border-radius: 3px;
  border-color: red;
  height: 35px;
}

.c1:focus {
  border-color: red;
  box-shadow: 0 0 20px red;
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 1000px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 2em;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 h3 {
  display: inline-block;
  margin: 0 1em;
}

<div>
  <div
    class="c0"
  >
    <span>
      <h3>
        Search:
      </h3>
      <input
        aria-label="search-input"
        class="c1"
        data-testid="search-input"
        type="search"
      />
    </span>
    <div
      class="c2"
    >
      <h3>
        Filtered users: 
        5
      </h3>
      <h3>
        Fetched users: 
        10
      </h3>
      <h3>
        Pages: 
        1
      </h3>
    </div>
  </div>
</div>
`;
